File: src/main/java/me/glorantq/aboe/chat/client/commands/ABOEClientCommand.java
package me.glorantq.aboe.chat.client.commands;

import net.minecraft.command.ICommand;
import net.minecraft.command.ICommandSender;
import net.minecraft.util.ChatComponentText;

public abstract class ABOEClientCommand implements ICommand {

    void sendMessage(ICommandSender sender, String message) {
        sender.addChatMessage(new ChatComponentText(message.replace("&", "\u00A7")));
    }

    @Override
    public boolean canCommandSenderUseCommand(ICommandSender p_71519_1_) {
        return true;
    }

    @Override
    public boolean isUsernameIndex(String[] p_82358_1_, int p_82358_2_) {
        return false;
    }

    @Override
    public int compareTo(Object o) {
        return 0;
    }
}

File: src/main/kotlin/me/glorantq/aboe/chat/client/commands/ABOEClientCommand.kt
package me.glorantq.aboe.chat.client.commands

import me.glorantq.aboe.chat.convertColours
import net.minecraft.command.ICommand
import net.minecraft.command.ICommandSender
import net.minecraft.util.ChatComponentText

abstract class ABOEClientCommand : ICommand {

    internal fun sendMessage(sender: ICommandSender, message: String) = sender.addChatMessage(ChatComponentText(message.convertColours()))
    override fun canCommandSenderUseCommand(p_71519_1_: ICommandSender): Boolean = true
    override fun isUsernameIndex(p_82358_1_: Array<String>, p_82358_2_: Int): Boolean = false
    override fun compareTo(other: Any?): Int = 0
}
